packages/compiler/core/checker.ts

```diff
@@ -3381,17 +3394,31 @@ export function createChecker(program: Program): Checker {
     if (program.literalTypes.has(value)) {
       return program.literalTypes.get(value)!;
     }
-    const kind =
-      typeof value === "string" ? "String" : typeof value === "number" ? "Number" : "Boolean";
-    const type: StringLiteralType | NumericLiteralType | BooleanLiteralType = createType({
-      kind,
-      value: value as any,
-    });
+    let type: StringLiteralType | NumericLiteralType | BooleanLiteralType;
 
+    switch (typeof value) {
+      case "string":
+        type = createType({ kind: "String", value });
+        break;
+      case "boolean":
+        type = createType({ kind: "Boolean", value });
+        break;
+      case "number":
+        type = createType({
+          kind: "Number",
+          value,
+          numericFlags: getNumericFlags(value, node as NumericLiteralNode),
+        });
+        break;
+    }
     program.literalTypes.set(value, type);
     return type;
   }
 
+  function getNumericFlags(value: number, node?: NumericLiteralNode) {
```

Comment:
The goal is to have that right?
```
-> means assignable

32    -> integer
32    -> real
32.5 -> real
```

With having it detect if its a real or integer it would let all of that work. If it is assumed to be a `real` then how does the first case work out. Or is  it just done during the checking of the relation that we check if the numeric literal value is an integer

Reply:
I think checking during relation is fine? I was somewhat confused with my original post as I see the type is still Number, but I think it's fine to check the Number value when it's important while doing type relations rather than storing the flags on the type.

Reply:
I guess I can move it back for now, but I think we'll need some kind of subtypes for numeric if we want to fully support int64 with BigInt right now we are loosing precision.